src/main/java/com/google/jenkins/plugins/computeengine/InstanceConfiguration.java

```diff
@@ -120,7 +126,19 @@ public class InstanceConfiguration implements Describable<InstanceConfiguration>
     private String machineType;
     private String numExecutorsStr;
     private String startupScript;
+
+    /**
+     * Use the {@link #provisioningType} field instead.
+     */
+    @Deprecated
     private boolean preemptible;
+
+    /**
+     * Succeeds {@link #preemptible}.
+     */
+    private ProvisioningType provisioningType;
```

Comment:
There is a bug with this code,

After saving the configuring with choosing a value for provisioning type (one of Standard, Sport VM, or Preemptible VM) - it works no issues, but if I try to update the configuration for a different parameter and open the configuration form again - this field is not setting back to the original value. So when I save the configuration without re-choosing the provisioning model, it gets set to `null`.

This is wrong because, then as per the `schedule` method logic, the provisioning falls to `standard` model as default, which is incorrect if it was preemptible / spot first time.

Reply:
Fixed after using the dropdownDescriptorSelector, configuration is saving, and stays correctly.